### PR TITLE
Editorial: Fix a couple of inconsistencies

### PR DESCRIPTION
**Pull request by linusg**
Found while implementing these, details in commit messages.

### CODE REVIEW

File: spec.html
Comment (michaelficarra) [inline spec.html:33775]: If we were to make this change, I would probably add

```suggestion
          1. If _tv_ is *NaN*, throw a *RangeError* exception.
          1. Assert: _tv_ is an integral Number.
```

File: spec.html
Comment (michaelficarra) [inline spec.html:39583]: I don't like these changes. Let's change `Array.prototype.join` to match this instead.

Comment (linusg) [reply spec.html]: Sure, done :+1:

**Review by michaelficarra — APPROVED** [review]

**Comment by linusg** [comment]
Rebased on latest main - is there anything preventing this from being merged?

**Comment by michaelficarra** [comment]
@linusg Waiting on review from another editor.

**Review by bakkot — APPROVED** [review]

**Comment by bakkot** [comment]
Backed out a couple changes the editors were less convinced of.

**Comment by linusg** [comment]
Fine by me, thanks for taking a look! Rebased and cleaned up the commit history a bit :+1:

**Review by bakkot — APPROVED** [review]

**Review by syg — APPROVED** [review]

**Pull request merged**
